google-ads-examples/src/main/java/com/google/ads/googleads/examples/basicoperations/AddResponsiveSearchAd.java

```diff
@@ -0,0 +1,160 @@
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param adGroupId the ad group ID.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId, long adGroupId) {
+    String adGroupResourceName = ResourceNames.adGroup(customerId, adGroupId);
+
+    // Sets a pinning to always choose this asset for HEADLINE_1. Pinning is optional; if no
+    // pinning is set, then headlines and descriptions will be rotated and the ones that
+    // perform best will be used more often.
+    AdTextAsset pinnedHeadline =
+        AdTextAsset.newBuilder()
+            .setText(StringValue.of("Cruise to Mars #" + System.currentTimeMillis()))
+            .setPinnedField(ServedAssetFieldType.HEADLINE_1)
+            .build();
+
+    // Creates the responsive search ad info.
+    ResponsiveSearchAdInfo responsiveSearchAdInfo =
+        ResponsiveSearchAdInfo.newBuilder()
+            .addHeadlines(pinnedHeadline)
+            .addHeadlines(createAdTextAsset("Best Space Cruise Line"))
+            .addHeadlines(createAdTextAsset("Experience the Stars"))
+            .addDescriptions(createAdTextAsset("Buy your tickets now"))
+            .addDescriptions(createAdTextAsset("Visit the Red Planet"))
+            .setPath1(StringValue.of("all-inclusive"))
+            .setPath2(StringValue.of("deals"))
+            .build();
+
+    // Wraps the info in an Ad object.
+    Ad ad =
+        Ad.newBuilder()
+            .setResponsiveSearchAd(responsiveSearchAdInfo)
+            .addFinalUrls(StringValue.of("http://www.example.com"))
+            .build();
+
+    // Builds the final ad group ad representation.
+    AdGroupAd adGroupAd =
+        AdGroupAd.newBuilder()
+            .setAdGroup(StringValue.of(adGroupResourceName))
+            .setStatus(AdGroupAdStatus.PAUSED)
+            .setAd(ad)
+            .build();
+
+    // Creates the operation.
+    AdGroupAdOperation operation = AdGroupAdOperation.newBuilder().setCreate(adGroupAd).build();
+
+    try (AdGroupAdServiceClient adGroupAdServiceClient =
+        googleAdsClient.getLatestVersion().createAdGroupAdServiceClient()) {
+      MutateAdGroupAdsResponse response =
```

Comment:
nit: add comment indicating this is the part that actually creates the ad group ad